libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -624,39 +624,49 @@ - (void)setOauthClientId:(NSString *)newClientId
 
 + (void)resetSessionCookie
 {
-    [self removeCookies:@[@"sid"]
-            fromDomains:@[@".salesforce.com", @".force.com", @".cloudforce.com"]];
-    [self addSidCookieForInstance];
+    __weak typeof(self) weakSelf = self;
+    [self removeCookiesFromDomains:@[@".salesforce.com", @".force.com", @".cloudforce.com"] withCompletion:^{
+        [weakSelf addSidCookieForInstance];
+    }];
 }
 
-+ (void)removeCookies:(NSArray *)cookieNames fromDomains:(NSArray *)domainNames
-{
-    NSAssert(cookieNames != nil && [cookieNames count] > 0, @"No cookie names given to delete.");
++ (void)removeCookiesFromDomains:(NSArray *)domainNames withCompletion:(nullable void(^)())completionBlock {
```

Comment:
Rather than introduce new interfaces here, I would prefer to do the following:

(
- Keep the `removeCookies:fromDomains:` interface.

_and_

- Fire and forget the calls to `[WKWebsiteDataStore removeDataOfTypes:forDataRecords:completionHandler:]`

_and_

- _Deprecate_ the method.

)

_OR_

- Do away with the method altogether.

The only thing that uses this method is `resetSessionCookie`, which should be amended to simply clear the shared process pool, because its current functionality does not work anyway: Adding the sid cookie back to `NSHTTPCookieStorage` is effectively a no-op for `WKWebView`.